### PR TITLE
crypto: Fix fallback key rotation so it works with wasm

### PR DESCRIPTION
**Pull request by uhoreg**
SystemTime doesn't work properly with wasm, so calculate milliseconds elapsed ourselves.

(otherwise failures like https://github.com/matrix-org/matrix-rust-sdk-crypto-wasm/actions/runs/8348688116/job/22851188836?pr=104 happen)

### CODE REVIEW

File: crates/matrix-sdk-crypto/src/olm/account.rs [hunk 1]
```diff
@@ -604,28 +603,24 @@ impl Account {
     ///
     /// [1]: https://signal.org/docs/specifications/x3dh/#publishing-keys
     fn fallback_key_expired(&self) -> bool {
-        const FALLBACK_KEY_MAX_AGE: Duration = Duration::from_secs(3600 * 24 * 7);
+        // 604800000 = 3600 * 24 * 7 * 1000 (one week in milliseconds)
+        let fallback_key_max_age: UInt = uint!(604800000);
```
Comment (uhoreg) [inline crates/matrix-sdk-crypto/src/olm/account.rs:607]: Rust doesn't seem to want to let this be a `const`, and the macro doesn't allow calculations inside it.

**Comment by poljar** [comment]
I think it would be better to fix this in Ruma, namely Ruma attempts to support WASM:

https://github.com/ruma/ruma/blob/f24cae17f50d140a0ff112cb3dc74a2053aa1df4/crates/ruma-common/src/time.rs#L24-L31

That's why the `MilisecondsSinceUnixEpoch::now()` call doesn't panic, but it returns `std::time::SytemTime` in the `to_system_time()` method. The `elapsed()` method of the `SystemTime` in the standard library panics under WASM.

It probably makes sense to modify Ruma so it returns a `SystemTime` that works under WASM, it's very unlikely that we're the only consumer that will bump into this problem.

Opened a PR for this here: https://github.com/ruma/ruma/pull/1747.

If you could confirm that the Ruma PR fixes your issue as well, that would be great.

**Comment by uhoreg** [comment]
After also applying https://github.com/ruma/ruma/pull/1749, the wasm issue is solved.

**Comment by uhoreg** [comment]
The issue was fixed in Ruma, so this PR is obsoleted by https://github.com/matrix-org/matrix-rust-sdk/pull/3264

**Pull request closed without merging**